services/network_config_manager/src/main/java/com/futurewei/alcor/netwconfigmanager/service/impl/GoalStatePersistenceServiceImpl.java

```diff
@@ -88,9 +85,13 @@
         return hostGoalStates;
     }
 
+    public Map<String, Neighbor.NeighborState> getNeighborStates(Set<String> resourceIds) throws Exception {
+        return resourceStateCache.getResourceStates(resourceIds);
+    }
+
     @Override
     @DurationStatistics
-    public synchronized boolean updateGoalState(String hostId, HostGoalState hostGoalState) throws Exception {
```

Comment:
Is this safe?

Reply:
Yes, outside this function, it already have synchronized.